lua/starfall/libs_sh/pvp.lua

```diff
@@ -0,0 +1,119 @@
+SF.RegisterLibrary( "pvp" )
+
+return function( instance )
+
+local pvp_library = instance.Libraries.pvp
+local checktype = instance.CheckType
+local player_methods = instance.Types.Player.Methods
+local ply_meta, punwrap = instance.Types.Player, instance.Types.Player.Unwrap
+
+local function getPly( this )
+    local ent = punwrap( this )
+
+    if ent:IsValid() then
+        return ent
+    else
+        SF.Throw( "Player is not valid.", 3 )
+    end
+end
+
+local function inFaction( ply )
+    local id = ply:GetNWString( "FactionID", nil )
+
+    if not id or id == "0" then
+        return false
+    end
+
+    return true
+end
+
+function pvp_library.getPvpers()
+    local pvpers = {}
+
+    for _, ply in pairs( player.GetHumans() ) do
+        if ply:GetNWBool( "CFC_PvP_Mode", false ) then
+            table.insert( pvpers, ply )
+        end
+    end
+
+    return pvpers
+end
+
+function pvp_library.getBuilders()
+    local builders = {}
+
+    for _, ply in pairs( player.GetHumans() ) do
+        if not ply:GetNWBool( "CFC_PvP_Mode", false ) then
+            table.insert( builders, ply )
+        end
+    end
+
+    return builders
+end
+
+-- isInPvp
+function pvp_library.playerIsInPvp( ply )
+    checktype( ply, ply_meta )
+
+    return getPly( ply ):GetNWBool( "CFC_PvP_Mode", false )
+end
+
+function player_methods:isInPvp()
+    checktype( self, ply_meta )
+
+    return getPly( self ):GetNWBool( "CFC_PvP_Mode", false )
+end
+
+-- isInBuild
+function pvp_library.playerIsInBuild( ply )
+    checktype( ply, ply_meta )
+
+    return not getPly( ply ):GetNWBool( "CFC_PvP_Mode", false )
```

Comment:
Is is possible to have this simply call the other method starfall method by not unpacking it?
```suggestion
    return not ply:isInPvp()
```

Reply:
And if thats the case, could this function and `playerIsInPvp` just call `ply:isInbuild` and `ply:IsInPvP` respectively?

Reply:
Unfortunately those methods are only defined on the server :(

Reply:
? They're defined in this file, by not calling `getPly` we're running the Starfall functions, not the lua ones

Reply:
I'm confused, you're gonna have to elaborate

Reply:
```lua
-- isInPvp
function pvp_library.playerIsInPvp( ply )
    checktype( ply, ply_meta )

    return ply:isInPvp()
end

function player_methods:isInPvp()
    checktype( self, ply_meta )

    return getPly( self ):GetNWBool( "CFC_PvP_Mode", false )
end

-- isInBuild
function pvp_library.playerIsInBuild( ply )
    checktype( ply, ply_meta )

    return ply:isInBuild()
end

function player_methods:isInBuild()
    checktype( self, ply_meta )

    return not self:isInPvp()
end
```
So the pvp_library functions just call the starfall player meta methods, and the player meta `isInBuild` just calls the player meta `isInPvp`
Important thing to note here is that I'm not using `getPly`, so the `isInbuild` and `isInPvp` functions I call on them will be the starfall ones (defined in this file), and not the lua ones (defined in factions)

I think you don't even need to `checktype` in `ply:isInbuild`, and the `isInPvp` check type will grab it, but I'd keep it just to be sure/consistent

Reply:
I'm not a big fan of this, I can probably just make a function at the top for getting the info